modules/loader/src/main/java/com/opengamma/strata/loader/csv/TradeCsvLoader.java

```diff
@@ -510,32 +512,39 @@ public boolean isKnownFormat(CharSource charSource) {
           case "FXSINGLE":
           case "FX SINGLE":
             if (tradeType == FxSingleTrade.class || tradeType == FxTrade.class || tradeType == Trade.class) {
-              trades.add(tradeType.cast(FxSingleTradeCsvPlugin.parse(row, info, resolver)));
+              trades.add(tradeType.cast(resolver.parseFxSingleTrade(row, info)));
             }
             break;
           case "FXSWAP":
           case "FX SWAP":
             if (tradeType == FxSwapTrade.class || tradeType == FxTrade.class || tradeType == Trade.class) {
-              trades.add(tradeType.cast(FxSwapTradeCsvPlugin.parse(row, info, resolver)));
+              trades.add(tradeType.cast(resolver.parseFxSwapTrade(row, info)));
             }
             break;
           case "FXVANILLAOPTION":
           case "FX VANILLA OPTION":
             if (tradeType == FxVanillaOptionTrade.class || tradeType == FxTrade.class || tradeType == Trade.class) {
-              trades.add(tradeType.cast(FxVanillaOptionTradeCsvPlugin.parse(row, info, resolver)));
+              trades.add(tradeType.cast(resolver.parseFxVanillaOptionTrade(row, info)));
             }
             break;
           case "CDS":
             if (tradeType == CdsTrade.class || tradeType == Trade.class) {
-              trades.add(tradeType.cast(CdsTradeCsvPlugin.parse(row, info, resolver)));
+              trades.add(tradeType.cast(resolver.parseCdsTrade(row, info)));
             }
             break;
           default:
-            failures.add(FailureItem.of(
-                FailureReason.PARSING,
-                "CSV file trade type '{tradeType}' is not known at line {lineNumber}",
-                typeRaw,
-                row.lineNumber()));
+            Optional<Trade> parsedOpt = resolver.parseOtherTrade(typeUpper, row, info);
+            if (parsedOpt.isPresent()) {
+              if (tradeType.isInstance(parsedOpt.get())) {
```

Comment:
If it isn't an instance should this add a failure?

Reply:
No, the type here is a filter, as per the other similar lines above,